/**:
  ros__parameters:
    # Topics
    pointCloudTopic: "/hesai/pandar"               # Point cloud data
    imuTopic: "/imu_data"                          # IMU data

    # Heading
    useImuHeadingInitialization: true             # if using GPS data, set to "true"

    # Export settings
    savePCD: false                                # https://github.com/TixiaoShan/LIO-SAM/issues/3
    savePCDDirectory: "/Downloads/LOAM/"          # starts and ends with "/". Warning: deletes folder

    # Lidar Settings
    N_SCAN: 32                                    # number of lidar channels
    Horizon_SCAN: 2000                            # lidar horizontal resolution
    ang_res_y: 1.0                                # vertical resolution of lidar
    lidar_type: 2                                 # 1 for ouster, 2 for hesai, 3 for velodyne
    timeField: "timestamp"                        # point timestamp field
    downsampleRate: 2                             # downsample data if too many points
    lidarMaxRange: 50.0
    lidarMinRange: 0.5
    NUM_OF_CAM: 2
    feature_enable: 0
    remove_noise: 0

    # Noise removal settings
    min_cluster_size: 10
    segment_valid_point_num: 5
    segment_valid_line_num: 3

    # IMU Settings
    imuAccNoise: 0.03015059
    imuGyrNoise: 0.42306179
    imuAccBiasN: 0.00198931
    imuGyrBiasN: 0.00015932
    imuGravity: 9.80

    # Extrinsics (lidar -> IMU)
    extrinsicTrans: [-0.0204, 0.0175, 0.0642]
    extrinsicRot: [-0.9741, 0.2262, 0.0008, -0.2261, -0.9740, 0.0147, 0.0041, 0.0141, 0.9999]
    extrinsicRPY: [0, -1, 0, -1, 0, 0, 0, 0, -1]

    # LOAM feature threshold
    edgeThreshold: 1.0
    surfThreshold: 0.1
    edgeFeatureMinValidNum: -1
    surfFeatureMinValidNum: 100

    # Voxel filter parameters
    odometrySurfLeafSize: 0.4
    mappingCornerLeafSize: 0.2
    mappingSurfLeafSize: 0.4

    # Robot motion constraint
    z_tollerance: 1000        # meters
    rotation_tollerance: 1000 # radians

    # CPU Params
    numberOfCores: 4
    mappingProcessInterval: 0.05

    # Surrounding map
    surroundingkeyframeAddingDistThreshold: 1.0
    surroundingkeyframeAddingAngleThreshold: 0.2
    surroundingKeyframeDensity: 2.0
    surroundingKeyframeSearchRadius: 50.0

    # Loop closure
    loopClosureEnableFlag: False
    surroundingKeyframeSize: 25
    historyKeyframeSearchRadius: 20.0
    historyKeyframeSearchTimeDiff: 30.0
    historyKeyframeSearchNum: 25
    historyKeyframeFitnessScore: 0.3

    # Visualization
    globalMapVisualizationSearchRadius: 1000.0
    globalMapVisualizationPoseDensity: 10.0
    globalMapVisualizationLeafSize: 1.0
